### PR TITLE
[WIP] Refactor transaction announcing+signing to be seperate re-usable functions 

### PR DESCRIPTION
**Pull request by ScaredIbis**
**_This still needs some testing but it would be good to get some opinions on the concept and whether this approach to the problem is suitable for the rest of the application:_**

**The Problem**
The Trezor device is going to be responsible for signing transactions, which we will then need to announce to the network seperately. From what I could see, the code currently doesn't support this scenario. Users of Trezor devices also will not have encrypted private keys within the wallet, so password confirmation doesn't apply when a user is finalising a new transaction from a Trezor account.

**The Solution**
This PR introduces an alternative to CheckPasswordDialog that is aware of the wallet sourceType (trezor, privateKey, seed) and renders the appropriate confirmation prompt to the user (password input, "Confirm via Trezor" button, etc).

This component is controlled through the vuex store via a new service function - `signTransaction`. This has simple usage and can be used from anywhere in the code to block the UI until the user confirms the signing of a transaction. `signTransaction` returns an instance of `SignedTransaction` to the caller.

```
const result = await signTransaction(Transaction, store);

// UI is blocked with a modal similar to CheckPasswordDialog until user actions the modal
// User will confirm with either password or trezor interaction depending on active wallet type
// User can also abort transaction confirmation by exiting from the modal

console.log(result.success) // true if transaction was successfully signed
console.log(result.error) //  error message from trezor OR 
// "User aborted transaction confirmation" if the user closes the modal or something similar
console.log(result.signedTransaction) // instance of SignedTransaction

new AppWallet(this.wallet).announceNormal(result.signedTransaction)
```

I have implemented this for Transfer Transaction types in `TransactionFormTs.ts` and it is being used for both Trezor and normal private key wallets.

### CODE REVIEW

**Comment by NemTechCN** [comment]
Very good to use! But it's better to show presentation after connect to Trezor ，you can use this.$Notice to add feedback.

File: src/core/model/AppWallet.ts [hunk 1]
```diff
@@ -389,23 +400,47 @@ export class AppWallet {
         )
     }
 
+    announceBonded(signedTransaction: SignedTransaction, node: string): void {
+        const transactionHttp = new TransactionHttp(node);
+        const listener = new Listener(node.replace('http', 'ws'), WebSocket)
+
+        listener.open().then(() => {
+            transactionHttp
+                .announce(signedTransaction)
```
Comment (evias) [inline src/core/model/AppWallet.ts]: this is announcing the signed aggregate bonded as a normal transaction?

I think what is needed here is to announce the funds lock transaction.

File: src/core/model/AppWallet.ts [hunk 2]
```diff
@@ -389,23 +400,47 @@ export class AppWallet {
...
+                .subscribe(x => console.log(x), err => console.error(err))
+
+            listener
+                .confirmed(this.simpleWallet.address)
+                .pipe(
+                filter((transaction) => transaction.transactionInfo !== undefined
+                    && transaction.transactionInfo.hash === signedTransaction.hash),
+                mergeMap(ignored => transactionHttp.announceAggregateBonded(signedTransaction)),
```
Comment (evias) [inline src/core/model/AppWallet.ts]: second time announcing the signed aggregate. This one can stay.

**Comment by ScaredIbis** [comment]
@evias and @decentraliser, I have played around a bit more and tested out sending transfer transactions from normal and multisig wallets, both seem to be working now. I have also made use of the nem2-hd-wallets package when extracting the extended public key from the Trezor. Would you mind having another at this when you have some time

**Comment by decentraliser** [comment]
Hi! When trying to announce a transfer I get the following error
`Error in render: "TypeError: Cannot read property 'address' of undefined" at new FormattedTransfer`

**Comment by ScaredIbis** [comment]
> Very good to use! But it's better to show presentation after connect to Trezor ，you can use this.$Notice to add feedback.

@NemTechCN sorry I don't quite understand what you mean. When you send the transaction with Trezor a different UI will show up and prompt the user to interact with the device. This is currently stubbed out at the moment but I am just introducing these changes now so they are compatible with the existing functionality and also when the Trezor transactions are implemented.

**Comment by decentraliser** [comment]
@evias LGTM, please have a look

**Comment by ScaredIbis** [comment]
hey @evias do you have some time soon to have another look at this? would be good to close it out

**Pull request merged**